complaint_search/es_interface.py

```diff
@@ -15,20 +15,130 @@
 _COMPLAINT_ES_INDEX = os.environ.get('COMPLAINT_ES_INDEX', 'complaint-index')
 _COMPLAINT_DOC_TYPE = os.environ.get('COMPLAINT_DOC_TYPE', 'complaint-doctype')
 
+_OPTIONAL_FILTERS = ("product", "issue", "company", "state", "zip_code", "timely", 
+    "company_response", "company_public_response", 
+    "consumer_consent_provided", "submitted_via", "tag")
+
+_OPTIONAL_FILTERS_PARAM_TO_ES_MAP = {
+    "product": "product.raw",
+    "sub_product": "sub_product.raw",
+    "issue": "issue.raw",
+    "sub_issue": "sub_issue.raw"
+}
```

Comment:
Currently `company_public_response.raw` and `consumer_consent_provided.raw` are currently not in the mappings, so it may not work at this moment.  @AdamZarger Can you add them to the mapping and I will do the same here for the API